### DISCUSSION TITLE
Feathers and Deno mongo

### DISCUSSION THREAD

**Discussion by vixalien** [feathersjs/feathers#2709] [Show and tell]
I made this [adapter](https://github.com/vixalien/feathers-mongo)  for feathers and `mongo` which is the de-facto Deno mongodb driver. I hoped someone would notice this as I'm not too suref everything looks right and maybe help me think out how to write test (im not very strong at testing).

Im also working on an `oak` adapter but that is proving to be more difficult than I thought.

**Comment by daffl** [comment]
This looks great, thank you for sharing! I was wondering if an adapter for Oak or plain Deno HTTP would be the best approach here. Either way, the real port is still a work in progress at https://github.com/feathersjs/feathers/pull/2531.

We haven't gotten back to it in a bit and I actually started wondering a few times now if it might make sense to turn Feathers core into a Deno first library (similar to https://github.com/feathersjs/hooks) with a different name and then just have FeathersJS use those modules published to npm.

**Comment by vixalien** [comment]
oh I think turning feathers into a core library is a great idea but I understand that there is https://deno.land/x/feathers and it works pretty well although it's kinda different from the NodeJS version. For once, it lacks the `teardown` method which causes TypeScript to freak out sometimes when it's used with NodeJS modules from https://esm.sh/@feathersjs/*

I also wonder why the other modules aren't published to Deno. Stuff like `@feathersjs/authentication-local` and others.

**Comment by vixalien** [comment]
hey @daffl. Here is the adapter for oak I've been talking about https://github.com/vixalien/feathers_oak/. currently misses auth.

**Reply by daffl** [reply]
This looks great! Do you want to hit me up [on Discord](https://discord.com/invite/qa8kez8QBx) to chat about how to put everything together? Having a MongoDB and HTTP adapter was definitely the things missing here.

The authentication modules haven't been ported because they rely on a lot of Node specific libraries (especially oAuth). I've also been thinking a bit about a websocket integration but didn't get too far on that either.

**Reply by vixalien** [reply]
hopped onto the server. had to (re) figure out discord first
